imod/formats/prj/prj.py

```diff
@@ -556,6 +556,24 @@ def _create_dataarray(
     elif values_valid:
         da = _create_dataarray_from_values(values_valid, headers_values)
 
+    da = apply_factor_and_addition(headers, da)
+    return da
+
+
+def apply_factor_and_addition(headers, da):
+    if not "layer" in da.coords:
+        factor = headers[0]["factor"]
+        addition = headers[0]["addition"]
+        da = da * factor
+        da = da + addition
+    else:
+        nlayers = len(da.coords["layer"])
+        for ilayer in range(nlayers):
+            layer = da.coords["layer"][ilayer].values[()]
+            factor = headers[ilayer]["factor"]
+            addition = headers[ilayer]["addition"]
+            da.loc[{"layer": layer}] = da.sel(layer=layer).values * factor
+            da.loc[{"layer": layer}] = da.sel(layer=layer).values + addition
```

Comment:
I think this works better with dask, as it prevents loading data in memory to apply factors and additions:

```suggestion
    layer, addition_values = list(zip([(i+1, header["addition"]) for i, header in enumerate(headers)]))
    factor_values = [header["factor"] for header in headers]
    addition = xr.DataArray(addition_values, coords = {"layer" : layer}, dims=("layer",))
    factor = xr.DataArray(factor_values, coords = {"layer" : layer}, dims=("layer",)
    da = da * factor + addition
```